### ISSUE TITLE
1.14 build broken with gcc 6.5.0

### ISSUE THREAD

**Issue by rofl0r** [hrydgard/ppsspp#16597]
### Game or games this happens in

None

### What area of the game / PPSSPP

-

### What should happen

-

### Logs

```
In file included from /src/build/ppsspp/ppsspp-1.14/ext/armips/Util/EncodingTable.h:4:0,
                 from /src/build/ppsspp/ppsspp-1.14/ext/armips/Core/Common.h:4,
                 from /src/build/ppsspp/ppsspp-1.14/ext/armips/Archs/Architecture.cpp:3:
/src/build/ppsspp/ppsspp-1.14/ext/armips/Util/FileClasses.h:6:20: fatal error: optional: No such file or directory
 #include <optional>
                    ^
compilation terminated.
make[2]: *** [ext/armips/CMakeFiles/armips.dir/Archs/Architecture.cpp.o] Error 1
```

### Platform

Linux / BSD

### Mobile phone model or graphics card

irrelevant

### PPSSPP version affected

v1.14

### Last working version

v1.13.1

### Graphics backend (3D API)

Other

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [ ] Try resetting settings or older versions and include if the issue is related.
- [ ] Try without any cheats and without loading any save states.
- [ ] Include logs or screenshots of issue.

**Comment by hrydgard** [comment]
Since the compile error is in an armips file, this would be an armips issue: https://github.com/Kingcom/armips

Seems armips simply requires a newer gcc compiler than that?

**Comment by rofl0r** [comment]
they probably dropped support for non-latest C++ standard recently. is it possible to turn off the assembler component ?

**Comment by hrydgard** [comment]
Not currently, but it would just take a CMake option and a smallish amount of ifdef-ing. PRs accepted.

**Comment by unknownbrackets** [comment]
Is there a motivation for still using gcc 6.5.0?

I'm worried you'd just run into another error later from us using something else in c++17.  I'm not really against an option to disable armips (although I think it's good to always have the debugger API available and would oppose a change to disable it by default anywhere.)

-[Unknown]

**Comment by rofl0r** [comment]
> Is there a motivation for still using gcc 6.5.0?

updating the default compiler version is always a big deal in a distro, as one often runs into lots of new issues. in my particular case / distro i'd have to update to at least gcc 11 which requires practically twice the amount of memory during compilation (this would mean that distro users with less than 4 GB ram would no longer be able to compile all components themselves), and would require an additional bootstrap step because gcc 11 and newer require a [C++11 compiler to build](https://gcc.gnu.org/install/prerequisites.html). previously we were able to build the base system with gcc 3.6.4 -> gcc 6.5.0, now it would look like gcc 3.4.6 -> gcc 6.5.0 -> gcc 11. going to all this trouble just to be able to compile a single package/library made by people that always want to use the latest C++ features seems  hard to justify. on the C side, people are much more conservative to ensure their projects can be built with 10+ year old compilers, many projects just recently started to adopt C99 which came out > 20 years ago. i've yet to see a program that requires C11.

**Comment by unknownbrackets** [comment]
I suppose clang is probably even harder in that regard.

We are sometimes motivated to use newer compiler features that make the code a lot simpler, or enable better performance without system specific hacks.  We've dropped old versions of Microsoft Visual C++ compilers and old unsupported toolchains like Symbian, and a part of it is that most people aren't testing in old compilers.  We've certainly run into multiple compiler bugs (either bad codegen or ICE) in the past, and likely codegen bugs just will go unnoticed when a tiny fraction of people use ancient compilers.

So I'll just warn you that the music will eventually stop for such an older system - although that doesn't mean it can't be more years from now.

You can try this branch.  I've created it, but am not interested enough to trouble myself to obtain and test an older compiler:
https://github.com/hrydgard/ppsspp/compare/master...unknownbrackets:ppsspp:armips-flag

-[Unknown]

**Comment by rofl0r** [comment]
many thanks @unknownbrackets - your fix works like a charm and ppsspp 1.14.1 builds again with armips disabled <3
